sources/web/datalab/polymer/test/item-list-test.ts

```diff
@@ -0,0 +1,211 @@
+  });
+
+  it('displays column names in the header row', () => {
+    testFixture.columns = ['col1', 'col2'];
+    Polymer.dom.flush();
+
+    // Column 0 is for the checkbox
+    assert(testFixture.$.header.children[1].innerText === 'col1');
+    assert(testFixture.$.header.children[2].innerText === 'col2');
+  });
+
+  it('selected items are selected', () => {
+    testFixture._selectItem(1);
+
+    assert(!isSelected(0), 'first item should not be selected');
+    assert(isSelected(1), 'second item should be selected');
+    assert(!isSelected(2), 'third item should not be selected');
+  });
+
+  it('selected items are returned', () => {
+    testFixture._selectItem(0);
+    testFixture._selectItem(2);
+
+    assert(JSON.stringify(testFixture.selectedIndices) === '[0,2]',
+        'first and third items should be selected');
+  });
+
+  it('can select/unselect all', () => {
+    const c = testFixture.$.header.querySelector('#selectAllCheckbox') as HTMLElement;
+    c.click();
+
+    assert(JSON.stringify(testFixture.selectedIndices) === '[0,1,2]');
+    assert(isSelected(0) && isSelected(1) && isSelected(2), 'all items should be selected');
+
+    c.click();
+
+    assert(JSON.stringify(testFixture.selectedIndices) === '[]');
+    assert(!isSelected(0) && !isSelected(1) && !isSelected(2), 'all items should be unselected');
+  });
+
+  it('clicking an item always selects it', () => {
+    const firstRow = getRow(0);
+    firstRow.click();
+
+    assert(isSelected(0), 'first item should be selected');
+
+    firstRow.click();
+
+    assert(isSelected(0), 'first item should still be selected');
+  });
+
+  it('selects only the clicked item if the checkbox was not clicked', () => {
+    const firstRow = getRow(0);
+    const secondRow = getRow(1);
+
+    firstRow.click();
+    secondRow.click();
+
+    assert(isSelected(1), 'only the second item should still be selected');
+  });
+
+  it('can do multi-selection using the checkboxes', () => {
+    const firstCheckbox = getCheckbox(0);
+    const thirdCheckbox = getCheckbox(2);
+
+    firstCheckbox.click();
+    assert(isSelected(0), 'first item should be selected');
+
+    thirdCheckbox.click();
+    assert(isSelected(0) && isSelected(2), 'both first and third items should be selected');
+    assert(!isSelected(1), 'second item should not be selected');
+  });
+
+  it('can do multi-selection using the ctrl key', () => {
+    const firstRow = getRow(0);
+    const thirdRow = getRow(2);
+
+    firstRow.click();
+    thirdRow.dispatchEvent(new MouseEvent('click', {
+      ctrlKey: true,
+    }));
+
+    assert(isSelected(0) && isSelected(2), 'both first and third items should be selected');
+    assert(!isSelected(1), 'second item should not be selected');
+
+    firstRow.dispatchEvent(new MouseEvent('click', {
+      ctrlKey: true,
+    }));
+
+    assert(!isSelected(0), 'first row should be unselected on ctrl click');
+  });
+
+  it('can do multi-selection using the shift key', () => {
```

Comment:
Done. Note the test should pass on all platforms. Even though you can't really make a ctrl+click on MacOS, you can still manually dispatch the event in javascript.